forem/forem
# Remove bootstrap classes : collapse hide show

## What type of PR is this? (check all applicable)


## Description

This PR focuses on removing bootstrap classes: `collapse`, `hide` and `show`.

These are removed by mainly using [details & summary](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_details) tags which provides the feature of expand and collapse.

Also, in `Feedback/Reports` the collapse functionality was not working earlier although the code was written so this PR fixes that too.

## Related Tickets & Documents



- Related Issue #18756

## QA Instructions, Screenshots, Recordings


**How to test?**

1. Sign in as admin locally
2. Visit http://localhost:3000/admin
3. Check expand/collapse feature in [Profile Fields](http://localhost:3000/admin/customization/profile_fields), [Flag Reactions](http://localhost:3000/admin/moderation/reports) and [Report #1](http://localhost:3000/admin/moderation/reports).

| Before  | After |
| ------------- | ------------- |
| <img width="791" alt="Screenshot 2023-01-02 at 8 59 01 PM" src="https://user-images.githubusercontent.com/9396084/210251726-6a69384b-248a-4d5d-8ad9-446f3aaaa0af.png"> | <img width="788" alt="Screenshot 2023-01-02 at 8 57 34 PM" src="https://user-images.githubusercontent.com/9396084/210251750-ad79009b-66ec-4029-99c4-dea5a7f830ff.png"> |
| <img width="791" alt="Screenshot 2023-01-02 at 8 59 10 PM" src="https://user-images.githubusercontent.com/9396084/210251807-66d5530c-c7da-4fa8-82da-e83ab0fdc036.png"> | <img width="792" alt="Screenshot 2023-01-02 at 8 57 59 PM" src="https://user-images.githubusercontent.com/9396084/210251787-d887f36b-dfca-4fde-bd05-2ed773451a9a.png"> |
| <img width="814" alt="Screenshot 2023-01-02 at 8 58 50 PM" src="https://user-images.githubusercontent.com/9396084/210251896-a4ae9d84-f8dc-43fb-9e29-597889b2c8dd.png"> | <img width="817" alt="Screenshot 2023-01-02 at 8 58 33 PM" src="https://user-images.githubusercontent.com/9396084/210251885-9e6cd9a5-8dbd-44cd-abc8-5c5ef3de81a9.png"> |
| <img width="796" alt="Screenshot 2023-01-02 at 8 59 28 PM" src="https://user-images.githubusercontent.com/9396084/210251987-bfc499ac-1327-460a-9a1e-2508a4eace35.png"> | <img width="795" alt="Screenshot 2023-01-02 at 8 58 19 PM" src="https://user-images.githubusercontent.com/9396084/210251967-a0ce191d-7449-4619-9f62-3fa41422a563.png"> |
| | <img width="790" alt="Screenshot 2023-01-02 at 8 57 46 PM" src="https://user-images.githubusercontent.com/9396084/210252036-f0f98fb9-b46a-4b55-bb76-79e6f6eae8f9.png"> |


### UI accessibility concerns?

The `details` and `summary` tags are not that great with accessibility but as per [this PR](https://github.com/forem/forem/pull/16620) its fine yo use them now as in future we will mostly removing the collapse/expand functionality.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
